src/evm-block-extractor/src/task/block_extractor.rs

```diff
@@ -134,6 +136,26 @@ impl<C: Client> BlockExtractor<C> {
             )
             .await??;
 
+            // Validate chain consistency, including new blocks sequence.
+            let validation_result = self.validate_chain(&evm_blocks).await;
```

Comment:
this function is only for collecting blocks, this additional logic don not belong here, should be performed somewhere else

Reply:
We collect blocks in loop, inserting them into DB on each iteration. The validation should be performed for each blocks sequence, before they added to DB. So, the validation logic should be inside the loop. I can only hide this code in separate fn.

Reply:
Please allow me to disagree. We have to imagine the entire flow as a pipe of steps, e.g.:
1. get local data info
2. fetch remote missing data
3. analyse fetched data
4. persist data
5. etc...

Each step should be performed by a dedicated function with a single resposibility, then all steps are aggregated by another function that calls and coordinatse them all, e.g.
```rust
fn do_all_job() {
    let local_data = get_local_data_info();
    let fetched_data = fetch_remote_missing_data(local_data);
    let clean_data = analyse_fetched_data(fetched_data);
    persist_data(clean_data);
    etc...
}
```

So each step is clear, simple, well encapsulated and easy to be tested in isolation.

Reply:
Split the collect all into smaller functions.